src/tpm2_wrap.c

```diff
@@ -172,6 +172,280 @@ int wolfTPM2_Init(WOLFTPM2_DEV* dev, TPM2HalIoCb ioCb, void* userCtx)
     return rc;
 }
 
+#ifndef WOLFTPM2_NO_HEAP
+WOLFTPM2_DEV *wolfTPM2_New(void)
+{
+    WOLFTPM2_DEV *dev = NULL;
+
+    dev = (WOLFTPM2_DEV *) XMALLOC(sizeof(WOLFTPM2_DEV), NULL,
+                                   DYNAMIC_TYPE_TMP_BUFFER);
+    if (dev == NULL) {
+        return NULL;
+    }
+
+    if (wolfTPM2_Init(dev, NULL, NULL) != TPM_RC_SUCCESS) {
+        return NULL;
+    }
+
+    return dev;
+}
+
+int wolfTPM2_Free(WOLFTPM2_DEV *dev)
+{
+    if (dev != NULL) {
+        wolfTPM2_Cleanup(dev);
+        XFREE(dev, NULL, DYNAMIC_TYPE_TMP_BUFFER);
+    }
+    return TPM_RC_SUCCESS;
+}
+
+WOLFTPM2_KEYBLOB* wolfTPM2_GetNewKeyBlob(void)
+{
+    WOLFTPM2_KEYBLOB* blob = NULL;
+
+    blob = (WOLFTPM2_KEYBLOB *) XMALLOC(sizeof(WOLFTPM2_KEYBLOB), NULL,
+                                      DYNAMIC_TYPE_TMP_BUFFER);
+    if (blob == NULL) {
+        return NULL;
+    }
+
+    XMEMSET(blob, 0, sizeof(WOLFTPM2_KEYBLOB));
+    return blob;
+}
+
+int wolfTPM2_CleanupKeyBlob(WOLFTPM2_KEYBLOB* blob)
+{
+    if (blob != NULL) {
+        XFREE(blob, NULL, DYNAMIC_TYPE_TMP_BUFFER);
+    }
+    return TPM_RC_SUCCESS;
+}
+
+WOLFTPM_API TPMT_PUBLIC* wolfTPM2_GetNewPublicTemplate(void)
+{
+    TPMT_PUBLIC* template = NULL;
+
+    template = (TPMT_PUBLIC *) XMALLOC(sizeof(TPMT_PUBLIC), NULL,
+                                       DYNAMIC_TYPE_TMP_BUFFER);
+    if (template == NULL) {
+        return NULL;
+    }
+
+    XMEMSET(template, 0, sizeof(TPMT_PUBLIC));
+    return template;
+}
+
+WOLFTPM_API int wolfTPM2_CleanupPublicTemplate(TPMT_PUBLIC* template)
+{
+    if (template != NULL) {
+        XFREE(template, NULL, DYNAMIC_TYPE_TMP_BUFFER);
+    }
+    return TPM_RC_SUCCESS;
+}
+
+WOLFTPM_API WOLFTPM2_KEY* wolfTPM2_GetNewKey(void)
+{
+    WOLFTPM2_KEY* key = NULL;
+
+    key = (WOLFTPM2_KEY *) XMALLOC(sizeof(WOLFTPM2_KEY), NULL,
+                                    DYNAMIC_TYPE_TMP_BUFFER);
+    if (key == NULL) {
+        return NULL;
+    }
+
+    XMEMSET(key, 0, sizeof(WOLFTPM2_KEY));
+    return key;
+}
+
+WOLFTPM_API int wolfTPM2_CleanupKey(WOLFTPM2_KEY* key)
+{
+    if (key != NULL) {
+        XFREE(key, NULL, DYNAMIC_TYPE_TMP_BUFFER);
+    }
+    return TPM_RC_SUCCESS;
+}
+
+WOLFTPM2_SESSION* wolfTPM2_GetNewSession(void)
+{
+    WOLFTPM2_SESSION* session = NULL;
+
+    session = (WOLFTPM2_SESSION *) XMALLOC(sizeof(WOLFTPM2_SESSION), NULL,
+                                    DYNAMIC_TYPE_TMP_BUFFER);
+    if (session == NULL) {
+        return NULL;
+    }
+
+    XMEMSET(session, 0, sizeof(WOLFTPM2_SESSION));
+    return session;
+}
+
+int wolfTPM2_CleanupSession(WOLFTPM2_SESSION* session)
+{
+    if (session != NULL) {
+        XFREE(session, NULL, DYNAMIC_TYPE_TMP_BUFFER);
+    }
+    return TPM_RC_SUCCESS;
+}
+#endif /* WOLFTPM2_NO_HEAP */
+
+WOLFTPM2_HANDLE* wolfTPM2_GetHandleRefFromKey(WOLFTPM2_KEY* key)
+{
+    if (key == NULL) {
+        return NULL;
+    }
+    return &(key->handle);
+}
+
+int wolfTPM2_OpenExistingDev(WOLFTPM2_DEV* dev)
```

Comment:
```
./configure --enable-swtpm CFLAGS="-DWOLFTPM2_NO_HEAP" && make

src/tpm2_wrap.c:299:5: error: no previous prototype for ‘wolfTPM2_OpenExistingDev’ [-Werror=missing-prototypes]
  299 | int wolfTPM2_OpenExistingDev(WOLFTPM2_DEV* dev)
      |     ^~~~~~~~~~~~~~~~~~~~~~~~
```

Reply:
Nice catch!! :+1:  I'll get that fixed up.